File: Question-3/game_session_system.js
class Player {
    constructor(name, skillLevel) {
        this.name = name;
        this.skillLevel = skillLevel;
        this.chatGroup = null;
    }
    joinChatGroup(group) {
        this.chatGroup = group;
        group.addPlayer(this);
    }
}
class ChatGroup {
    constructor(skillLevel) {
        this.skillLevel = skillLevel;
        this.players = [];
    }
    addPlayer(player) {
        if (player.skillLevel === this.skillLevel) {
            this.players.push(player);
            console.log(`${player.name} joined the chat group for skill level ${this.skillLevel}`);
        } else {
            console.log(`${player.name} does not match the skill level of this group.`);
        }
    }
    checkBribery() {
      
        const briberyDetected = Math.random() < 0.1; 
        return briberyDetected ? "Bribery detected!" : "No bribery detected.";
    }
}

const player1 = new Player("Alice", 5);
const player2 = new Player("Bob", 5);
const player3 = new Player("Charlie", 6);
const group5 = new ChatGroup(5);
const group6 = new ChatGroup(6);
player1.joinChatGroup(group5);
player2.joinChatGroup(group5);
player3.joinChatGroup(group6);
console.log(group5.players);
console.log(group6.players);
console.log(group5.checkBribery());

File: Question-3/game_session_system.kt

data class Player(val name: String, val skillLevel: Int) {
    var chatGroup: ChatGroup? = null

    fun joinChatGroup(group: ChatGroup) {
        chatGroup = group
        group.addPlayer(this)
    }
}

class ChatGroup(val skillLevel: Int) {
    private val players = mutableListOf<Player>()

    fun addPlayer(player: Player) {
        if (player.skillLevel == skillLevel) {
            players.add(player)
            println("${player.name} joined the chat group for skill level $skillLevel")
        } else {
            println("${player.name} does not match the skill level of this group.")
        }
    }
}

    fun checkBribery(): String {
    
        val briberyDetected = Random.nextDouble() < 0.1 // 10% chance of detecting bribery
        return if (briberyDetected) "Bribery detected!" else "No bribery detected.”
    }
